### DISCUSSION TITLE
Is there a way to lint for and prevent `_ is False` (identity comparison)? (stricter E712)

### DISCUSSION THREAD

**Discussion by jbcpollak** [astral-sh/ruff#12584] [Q&A]
I would like to enforce that checks for True/False identity always take the form `if _` rather than `if _ is True`. E712 allows the `is` form. Is there an enhanced check I could do?

**Comment by MichaReiser** [comment]
I don't think such a rule exists today. @AlexWaygood any thoughts?

**Comment by AlexWaygood** [comment]
E712 allows `_ is False` comparisons to avoid false positives, since there are _some_ occasions when you do want to check that something is exactly the `False` constant rather than just something that has a false boolean value. However, I agree that those occasions are pretty rare, so I think it's reasonable to want a stricter check.

Maybe instead of adding a new rule, we could just make E712 configurable. The default could be the current behaviour (lenient, avoids false positives), but we could have a setting that makes it stricter and also disallows identity comparisons to the `True` and `False` constants.

**Reply by jbcpollak** [reply]
if a new or configurable E712 is an option, it seems like it would address https://github.com/astral-sh/ruff/issues/4560 as well.

**Comment by jbcpollak** [comment]
That would be great - I'd be happy with either a new rule or making E712 configurable.

Is their precedent for configurable rules in Python? It's common in JS linting tools but I don't think I've seen it in Python before.

**Reply by AlexWaygood** [reply]
We have quite a few configurable rules in Ruff. Most of our `flake8-pytest-style` rules [are configurable](https://docs.astral.sh/ruff/settings/#lint_flake8-import-conventions_extend-aliases), and for most of our `pydocstyle` rules you can specify the docstring convention you want to enforce, etc.

**Reply by jbcpollak** [reply]
Ah. I hadn't noticed. Either way works for me!
